### DISCUSSION TITLE
请问如果想给所有的选择类组件加上自定义的查询接口该如何操作

### DISCUSSION THREAD

**Discussion by kkodeok** [alibaba/formily#2112] [Q&A 2.x] [answered]
例如给select传参的时候传入一些api及post body为参数，然后可以直接实现请求该接口并且将结果设置为select的可选值，类似这种操作有没有官方建议的二次开发方案呢？

**Comment by janryWang** [comment, marked as answer]
```
effects:()=>{
   onFieldInit("*",(field)=>{
      //找到Select,给组件塞属性
   })
}
```

**Reply by kkodeok** [reply]
谢谢解答，请问如果我使用JSON Schema的模式的话这个同样是适用吗，以及您这里的塞属性是指塞jSX模式下的select中的DataSource吗？

**Reply by janryWang** [reply]
这个是顶层控制的，不约束是不是JSON-Schema

**Reply by kkodeok** [reply]
好的，之前有个思路是在表单设计器里面看到有配置响应器的操作里面有动作响应可以绑定动作，请问这个动作支持引用设置自定义的方法吗，比如这里的fetch改成umi-request这种？

**Reply by janryWang** [reply]
你注册scope就能消费

**Reply by kkodeok** [reply]
十分抱歉，请问这个有样例或者文档说明吗  对于注册scope不是很理解

**Reply by kkodeok** [reply]
已经在文档找到了相关定义，谢谢解答    非常感谢
